src/kudu/consensus/consensus_queue.cc

```diff
@@ -1051,6 +1062,51 @@ void PeerMessageQueue::PromoteIfNeeded(TrackedPeer* peer, const TrackedPeer& pre
   }
 }
 
+bool PeerMessageQueue::BasicChecksOKToTransferUnlocked(
+    const TrackedPeer& peer, RaftPeerPB **peer_pb_ptr) {
+  DCHECK(queue_lock_.is_locked());
+
+  // This check is redundant for ResponseFromPeer common path
+  if (PREDICT_FALSE(queue_state_.state != kQueueOpen)) {
+    return false;
+  }
+
+  if (queue_state_.mode != PeerMessageQueue::LEADER ||
```

Comment:
upstream has an additional condition here, minor but maybe worth adding:

```local_peer_pb_.permanent_uuid() == peer.uuid() ||```

Reply:
at this point the tracked peer has already been gotten by getting designated_peer.uuid(), so it is redundant and correct by construction.

Reply:
OK, I didn't trace it all the way back, just noticed this had been added since we diverged.